src/encoded/helpers/upload_credentials.py

```diff
@@ -46,7 +138,17 @@ def _get_token(self, policy):
             print('Warning: ', ecp)
             return None
 
-    def external_creds(self):
+    def _check_external_policy(self, allow, bucket_list_path):
+        if allow and bucket_list_path:
+            external_policy = _get_external_bucket_policy(
+                bucket_list_path,
+                retry=True
+            )
+            if external_policy:
+                self._external_policy = external_policy
+
+    def external_creds(self, allow=False, bucket_list_path=None):
```

Comment:
If you're fixing stuff already then maybe change 'allow' to more descriptive name.